unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -850,7 +850,7 @@ public int compareTo(@Nonnull final UTF8String other) {
         return res;
       }
     }
-    return numBytes - other.numBytes;
+    return numBytes < other.numBytes ? -1 : (numBytes > other.numBytes ? 1 : 0);
```

Comment:
What's the rationale? I don't think this makes the code more readable. If you want to return just one of `-1, 0, 1` you could use `Integer.signum`, though it's not necessary.